Samples/DirectMLNpuInference/main.cpp

```diff
@@ -13,11 +13,36 @@
 
 using Microsoft::WRL::ComPtr;
 
-void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** commandQueueOut, IDMLDevice** dmlDeviceOut) {
-    // Whether to skip adapters which support Graphics in order to target NPU for testing
-    bool forceComputeOnlyDevice = true;
-    bool forceGenericMLDevice = false;
-    
+void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** commandQueueOut, IDMLDevice** dmlDeviceOut)
+{
+    // Useful Flags to chage to allow, require, and disallow certain attributes of devices for testing.
+    const bool allowGraphicsCapabilities = false;
+    const bool requireComputeDevice = true;
+    const bool requireGenericMLDevice = false;
+
+    // Populate helper structures based on above flags
+    std::vector<GUID> dxGuidAllowedAttributes = {};
+    std::vector<GUID> dxGuidRequiredAttributes = {};
+    std::vector<GUID> dxGuidDisallowedAttributes = {};
```

Comment:
Do we need both allowed and required, if we are going to filter out anything in allowed that's not in required?